packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -694,22 +694,20 @@ export class EqualTo extends Op {
     this.assertMinStackLen(stack, 2);
     const last = stack.pop();
     const prev = stack.pop();
-    if (typeof last === typeof prev) {
-      if (typeof last === "bigint") {
-        if (last === prev) {
-          stack.push(BIGINT1);
-        } else {
-          stack.push(BIGINT0);
-        }
+    if (typeof last !== typeof prev) {
+      throw new TealError(ERRORS.TEAL.INVALID_TYPE);
+    } else if (typeof last === "bigint") {
```

Comment:
else not needed here, because we throw exception.
So we can do a normal block:
```
if (typeof last !== typeof prev)
        throw new TealError(ERRORS.TEAL.INVALID_TYPE);
if (typeof last === "bigint") {
   ...
```

Reply:
ok